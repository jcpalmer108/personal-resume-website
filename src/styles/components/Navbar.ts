import styled from "styled-components";
import {
  tabletBreakpoint,
  redstone,
  silver_polish,
  desktopBreakpoint,
} from "../variables";

export const Logo = styled.img`
  height: 40px;

  @media (min-width: ${desktopBreakpoint}px) {
    height: 55px;
  }
`;

export const Hamburger = styled.div`
  border: 1px solid ${redstone};
  height: 40px;
  width: 40px;
  display: flex;

  .hamburger-react {
    top: -3.5px;
    left: -3.5px;
  }

  @media (min-width: ${tabletBreakpoint}px) {
    display: none;
  }
`;

export const UpperMenu = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 100px;
  justify-content: space-between;
  padding: 0px 20px;

  @media (min-width: ${tabletBreakpoint}px) {
    padding: 0px 25px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding: 0px 60px;
    height: 105px;
  }
`;

export const LowerMenu = styled.div`
  min-height: 320px;
  height: calc(100vh - 95px);
  display: flex;
  flex-direction: column;
`;

export const MobileMenu = styled.div`
  flex-grow: 1;
  display: flex;
  padding: 40px 0px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const MobilePages = styled.a`
  height: 100%;
  display: flex;
  align-items: center;
  text-decoration: none;
  color: black;
`;

export const SocialsWrapper = styled.div`
  background: white;
  justify-content: space-around;
  display: flex;
  height: 60px;
`;

export const Socials = styled.a`
  border-left: 1px solid ${silver_polish};
  border-bottom: 1px solid ${silver_polish};
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;

  &:first-child {
    border-left: none;
  }
`;

export const Icon = styled.img`
  height: 26px;
`;

export const DesktopMenu = styled.div`
  display: none;
  line-spacing: 3px;

  @media (min-width: ${tabletBreakpoint}px) {
    display: flex;
    justify-content: space-between;
    align-items: right;
  }
`;

export const DesktopPages = styled.a`
  text-decoration: none;
  color: black;
  padding-left: 24px;
  font-family: Gilroy-Medium;
  font-size: 12px;
  transition: all 500ms ease;

  &:first-child {
    padding-left: none;
  }

  @media (min-width: ${tabletBreakpoint}px) {
    &:hover {
      font-size: 14px;
      color: ${redstone};
      transition: all 500ms ease;
    }
  }

  @media (min-width: ${desktopBreakpoint}px) {
    font-size: 14px;
    padding-left: 40px;

    &:hover {
      font-size: 16px;
      color: ${redstone};
      transition: all 500ms ease;
    }
  }
`;
