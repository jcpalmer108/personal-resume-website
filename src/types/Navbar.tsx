import { MenuItem } from "./global";

export type Link = {
  key: string;
  label: string;
  url: string;
};

export type NavbarProps = {
  menu: MenuItem[];
};
