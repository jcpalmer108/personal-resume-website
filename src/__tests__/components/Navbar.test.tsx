import { render, screen } from "@testing-library/react";
import Navbar from "../../components/Navbar";

const params = [
  {
    key: "one",
    label: "One",
  },
  {
    key: "two",
    label: "Two",
  },
];

describe("Navbar", () => {
  test("renders if only required params are passed in", () => {
    render(<Navbar menu={params} />);
    expect(screen.getByTestId("Navbar")).toMatchSnapshot();
    expect(screen.getByTestId("Desktop Menu").childNodes).toHaveLength(
      params.length
    );
    expect(screen.getByTestId("Mobile Menu").childNodes).toHaveLength(
      params.length
    );
    expect(screen.getByTestId("Socials")).toBeTruthy();
    expect(screen.getByTestId("Hamburger")).toBeTruthy();
  });
});
