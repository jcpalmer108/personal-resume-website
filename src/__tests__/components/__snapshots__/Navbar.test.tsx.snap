// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders if only required params are passed in 1`] = `
<div
  data-testid="Navbar"
  style="opacity: 0; max-height: 105px; overflow: hidden;"
>
  <div
    class="sc-ftvSup fqXeAh"
  >
    <img
      alt="logo"
      class="sc-gKXOVf bgZQNE"
      src="jennapalmer-dark.svg"
    />
    <div
      class="sc-breuTD bQjhPd"
      data-testid="Desktop Menu"
    >
      <a
        class="sc-ksZaOG hPOIwE"
        href="#one"
      >
        ONE
      </a>
      <a
        class="sc-ksZaOG hPOIwE"
        href="#two"
      >
        TWO
      </a>
    </div>
    <div
      class="sc-iBkjds cMkmwA"
      data-testid="Hamburger"
    >
      <div
        aria-expanded="false"
        class="hamburger-react"
        role="button"
        style="cursor: pointer; height: 48px; position: relative; transition: 0.5s cubic-bezier(0, 0, 0, 1); user-select: none; width: 48px; outline: none;"
        tabindex="0"
      >
        <div
          style="background: rgb(229, 107, 111); height: 2px; left: 14px; position: absolute; width: 20px; top: 17px; transition: 0.5s cubic-bezier(0, 0, 0, 1); transform: none;"
        />
        <div
          style="background: rgb(229, 107, 111); height: 2px; left: 14px; position: absolute; width: 20px; top: 23px; transition: 0.25s cubic-bezier(0, 0, 0, 1); transform: none;"
        />
        <div
          style="background: rgb(229, 107, 111); height: 2px; left: 14px; position: absolute; width: 20px; top: 29px; transition: 0.5s cubic-bezier(0, 0, 0, 1); transform: none;"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-papXJ jmtnqI"
  >
    <div
      class="sc-jqUVSM emVZew"
      data-testid="Mobile Menu"
    >
      <a
        class="sc-kDDrLX doDXhI"
        href="#one"
      >
        ONE
      </a>
      <a
        class="sc-kDDrLX doDXhI"
        href="#two"
      >
        TWO
      </a>
    </div>
    <div
      class="sc-iqcoie lihoFq"
      data-testid="Socials"
    >
      <a
        class="sc-crXcEl gmDjrd"
        href="https://www.github.com/jcpalmer108/"
      >
        <img
          alt="github"
          class="sc-evZas leZbCs"
          src="[object Object]"
        />
      </a>
      <a
        class="sc-crXcEl gmDjrd"
        href="https://leetcode.com/darthToaster/"
      >
        <img
          alt="leetcode"
          class="sc-evZas leZbCs"
          src="[object Object]"
        />
      </a>
      <a
        class="sc-crXcEl gmDjrd"
        href="https://www.linkedin.com/in/jenna-palmer/"
      >
        <img
          alt="linkedin"
          class="sc-evZas leZbCs"
          src="[object Object]"
        />
      </a>
      <a
        class="sc-crXcEl gmDjrd"
        href="https://twitter.com/jpalmerproject/"
      >
        <img
          alt="twitter"
          class="sc-evZas leZbCs"
          src="[object Object]"
        />
      </a>
      <a
        class="sc-crXcEl gmDjrd"
        href="mailto:jenna@jennapalmer.info"
      >
        <img
          alt="email"
          class="sc-evZas leZbCs"
          src="[object Object]"
        />
      </a>
    </div>
  </div>
</div>
`;
